webapp/src/components/Countdown/Countdown.js

```diff
@@ -0,0 +1,139 @@
+
+    this.state = {
+      days: 0,
+      hours: 0,
+      min: 0,
+      sec: 0
+    }
+  }
+
+  componentDidMount() {
+    // update every second
+    this.interval = setInterval(() => {
+      const date = this.calculateCountdown(this.props.date)
+      date ? this.setState(date) : this.stop()
+    }, 1000)
+  }
+
+  componentWillUnmount() {
+    this.stop()
+  }
+
+  calculateCountdown(endDate) {
+    let diff = (Date.parse(new Date(endDate)) - Date.parse(new Date())) / 1000
+
+    // clear countdown when date is reached
+    if (diff <= 0) return false
+
+    const timeLeft = {
+      years: 0,
+      days: 0,
+      hours: 0,
+      min: 0,
+      sec: 0
+    }
+
+    // calculate time difference between now and expected date
+    if (diff >= 365.25 * 86400) {
+      // 365.25 * 24 * 60 * 60
+      timeLeft.years = Math.floor(diff / (365.25 * 86400))
+      diff -= timeLeft.years * 365.25 * 86400
+    }
+    if (diff >= 86400) {
+      // 24 * 60 * 60
+      timeLeft.days = Math.floor(diff / 86400)
+      diff -= timeLeft.days * 86400
+    }
+    if (diff >= 3600) {
+      // 60 * 60
+      timeLeft.hours = Math.floor(diff / 3600)
+      diff -= timeLeft.hours * 3600
+    }
+    if (diff >= 60) {
+      timeLeft.min = Math.floor(diff / 60)
+      diff -= timeLeft.min * 60
+    }
+    timeLeft.sec = diff
+
+    return timeLeft
+  }
+
+  stop() {
+    clearInterval(this.interval)
+  }
+
+  addLeadingZeros(value) {
+    value = String(value)
+    while (value.length < 2) {
+      value = '0' + value
+    }
+    return value
+  }
+
+  render() {
+    const countDown = this.state
+
+    return (
+      <div className="Countdown">
+        <span className="Countdown-col">
+          <span className="Countdown-col-element">
+            <p>{this.addLeadingZeros(countDown.days)}</p>
+            <span>
+              {countDown.days === 1 ? t('global.day') : t('global.days')}
+            </span>
+          </span>
+        </span>
+
+        <span className="Countdown-col">
+          <span className="Countdown-col-element">
+            <p>{this.addLeadingZeros(countDown.hours)}</p>
+            <span>{t('global.hours')}</span>
+          </span>
+        </span>
+
+        <span className="Countdown-col">
+          <span className="Countdown-col-element dots">
+            <p>:</p>
+          </span>
+        </span>
+
+        <span className="Countdown-col">
+          <span className="Countdown-col-element">
+            <p>{this.addLeadingZeros(countDown.min)}</p>
+            <span>{t('global.minutes')}</span>
+          </span>
+        </span>
+
+        <span className="Countdown-col">
+          <span className="Countdown-col-element dots">
+            <p>:</p>
+          </span>
+        </span>
+
+        <span className="Countdown-col">
+          <span className="Countdown-col-element">
+            <p>{this.addLeadingZeros(countDown.sec)}</p>
+            <span>{t('global.seconds')}</span>
+          </span>
+        </span>
```

Comment:
This HTML looks super weird to me, too many `span`s for no reason? It's not that's bad or anything but the building block for `xxx-col` seems like it should be a div?

cc: @cazala

Reply:
Just something to have in mind. I copied and paste it.

I could revisit it. It was done to speed up the development. My bad